### PR TITLE
AR Refactor / Bcomps

### PR DESCRIPTION
**Pull request by cameron-freshworks**
Issue #: /bcgov/entity#1424 & /bcgov/entity#1425

Description of changes:

* Refactored the Annual Report component to support a BCOMPs filing while keeping the COOPs filing unchanged.

* Created a new Address Summary Component to display summarized, un-editable address data.

* Updated Unit Testing for Annual Report

* New Unit Testing for the summaryOfficeAddresses component.

* Designs for reference : 
https://invis.io/MHTDPV2Z4GS

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: coops-ui/src/components/Common/SummaryOfficeAddresses.vue [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+          </div>
+        </li>
+      </div>
+      <div v-else>
+        <li class="address-list-container">
+          <div class="meta-container">
+            <label>Records Office</label>
+            <div class="meta-container__inner">
+              <span id="sameAsRegistered">
+                Same as Registered Office
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply coops-ui/src/components/Common/SummaryOfficeAddresses.vue]: Thats a good question. This summary component contains the same functionality as the OfficeAddress component, but i will ask business if they still want to 'ShortHand' this for this summary screen. Great question!

File: coops-ui/src/components/Common/SummaryOfficeAddresses.vue [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  @Prop({ default: null })
+  private recordsAddress: BaseAddressObjIF
+
+  // Entity Enum
+  private EntityTypes: {} = EntityTypes
+}
+</script>
+
+<style lang="scss" scoped>
+  @import '../../assets/styles/theme.scss';
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply coops-ui/src/components/Common/SummaryOfficeAddresses.vue]: Good catch, i believe Severin just updated this work in his latest PR. I will change this so it's inline with his most recent refactor to theme imports. Thanks!

File: coops-ui/src/components/Common/SummaryOfficeAddresses.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply coops-ui/src/components/Common/SummaryOfficeAddresses.vue]: So as per the designs, this functionality to shorthand the addresses when similar is still in play, though i'm not sure if it is a documented requirement.

File: coops-ui/src/views/AnnualReport.vue [hunk 3]
```diff
@@ -36,113 +36,140 @@
...
-                    :addresses.sync="addresses"
-                    :registeredAddress.sync="registeredAddress"
-                    :recordsAddress.sync="recordsAddress"
-                    @modified="officeModifiedEventHandler($event)"
-                    @valid="addressFormValid = $event"
-                  />
-                </section>
+            <article>
+            <!-- Annual Report Date ( COOP ) -->
+              <article
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: "Article" is an HTML5 semantic element, and I think it doesn't make sense to nest one article inside another. Perhaps the outer one should be a section? See https://www.w3schools.com/html/html5_semantic_elements.asp
Or, perhaps the outer article doesn't need to exist?

File: coops-ui/src/views/AnnualReport.vue [hunk 4]
```diff
@@ -36,113 +36,140 @@
...
+                v-if="entityFilter(EntityTypes.COOP)"
+              >
+                <header>
+                  <h1 id="AR-header">File {{ ARFilingYear }} Annual Report
+                    <span class="font-italic" v-if="reportState">- {{ reportState }}</span>
+                  </h1>
+                  <p>Please verify or change your Office Addresses and Directors.</p>
+                </header>
+
+                <div v-if="isAnnualReportEditable">
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: As discussed earlier, this div is not used for any semantic purpose (ie, has no class and is therefore not used for layout or style purposes) but it still rendered (when its conditional is true).
Better code would be to use a "template" tag, which is never rendered, so you save one level of nesting in the DOM.

File: coops-ui/src/views/AnnualReport.vue [hunk 5]
```diff
@@ -203,31 +235,76 @@
...
-// action constants
-const APPOINTED = 'appointed'
-const CEASED = 'ceased'
-const NAMECHANGED = 'nameChanged'
-const ADDRESSCHANGED = 'addressChanged'
+// Mixins
+import { DateMixin, EntityFilterMixin } from '@/mixins'
+
+// Constants
+import { APPOINTED, CEASED, NAMECHANGED, ADDRESSCHANGED } from '@/constants'
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: Love it :)

File: coops-ui/src/views/AnnualReport.vue [hunk 6]
```diff
@@ -283,7 +362,7 @@ export default {
       // other local properties
       filingId: null,
       loadingMessage: 'Loading...', // initial generic message
-      filingData: [],
+      filingData: [] as Array<FilingData>,
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: Nice! :)

File: coops-ui/src/views/AnnualReport.vue [hunk 7]
```diff
@@ -358,6 +441,8 @@ export default {
     } else {
       // else just load new page
       this.loadingMessage = `Preparing Your ${this.ARFilingYear} Annual Report`
+      // TODO: This filing code needs to be updated to ANNUAL_REPORT_BC when Payments codes have been updated.
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: How are you tracking that this future work gets done? Ticket #?

File: coops-ui/src/components/Common/SummaryOfficeAddresses.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply coops-ui/src/components/Common/SummaryOfficeAddresses.vue]: Thanks for fixing this.

File: coops-ui/src/components/Common/SummaryOfficeAddresses.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply coops-ui/src/components/Common/SummaryOfficeAddresses.vue]: Thanks for confirming this.
Should some test notes (or updated Acceptance Criteria) be added to the ticket?

File: coops-ui/src/views/AnnualReport.vue [hunk 8]
```diff
@@ -203,31 +235,76 @@
...
+              <div v-on="on" class="d-inline">
+                <v-btn
+                  id="ar-file-pay-bc-btn"
+                  color="primary"
+                  large
+                  :disabled="!validated"
+                  :loading="filingPaying"
+                  @click="onClickFilePay"
+                >
+                  File & Pay
```
Comment (severinbeauvais) [inline coops-ui/src/views/AnnualReport.vue]: Change & to &amp; .

File: schemas/src/registry_schemas/schemas/annual_report.json [hunk 9]
```diff
@@ -11,7 +11,6 @@
         "annualReport": {
             "type": "object",
             "required": [
-                "annualGeneralMeetingDate",
```
[the remark this answers is not in the document]
Comment (cameron-freshworks) [reply schemas/src/registry_schemas/schemas/annual_report.json]: 100%, i have reverted this back and will leave for BE PR Merge to update this. Thanks for the heads up!

File: coops-ui/src/views/AnnualReport.vue (hunk 3 above)
Comment (cameron-freshworks) [reply coops-ui/src/views/AnnualReport.vue]: Great suggestion 👍  I updated and wrapped the articles in a <section>

File: coops-ui/src/views/AnnualReport.vue (hunk 7 above)
Comment (cameron-freshworks) [reply coops-ui/src/views/AnnualReport.vue]: I will create a ticket right now!

Comment (cameron-freshworks) [reply coops-ui/src/views/AnnualReport.vue]: #2054

Placeholder ticket, until we can get the details from business regarding all the various codes 👍

File: coops-ui/src/views/AnnualReport.vue (hunk 3 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Great!

File: schemas/src/registry_schemas/schemas/annual_report.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (severinbeauvais) [reply schemas/src/registry_schemas/schemas/annual_report.json]: I see the change is restored. Thanks.

File: coops-ui/src/views/AnnualReport.vue (hunk 4 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Changed!

File: coops-ui/src/views/AnnualReport.vue (hunk 7 above)
Comment (severinbeauvais) [reply coops-ui/src/views/AnnualReport.vue]: Perfect, thanks!

**Pull request merged**